nips2016_NAMPI/process/commentProcess.js

```diff
@@ -0,0 +1,133 @@
+        };
+        return or3client.or3request( or3client.mailUrl, reviewer_mail, 'POST', token );
+      })
+    };
+
+    var getPCEmails = function(){
+      console.log('get PC emails initiated')
+      return or3client.or3request(or3client.grpUrl+'?id=NIPS.cc/2016/workshop/NAMPI/pcs',{},'GET',token)
+      .then(result=>{      
+        var pcs = result.groups[0].members;
+        var signatureIdx = pcs.indexOf(note.signatures[0]);
+        console.log('pcs before filter:',pcs);
+        if(signatureIdx>-1){
+          pcs.splice(signatureIdx,1);
+        };
+        console.log('pcs after filter:',pcs);
+        var pc_mail = {
+          "groups": pcs,
+          "subject": "Private comment posted to a paper: \"" + note.content.title + "\"",
+          "message": "A submission to "+ conference+" has received a comment for the Program Chairs. \n\nTitle: "+note.content.title+"\n\nComment: "+note.content.comment+"\n\nTo view the comment, click here: "+baseUrl+"/forum?id=" + note.forum
+        };
+        return or3client.or3request( or3client.mailUrl, pc_mail, 'POST', token );
+      })
+    };
+
+    var getCommentEmails = function(replytoNoteSignatures){
+      console.log('replytoNoteSignatures before filter:',replytoNoteSignatures);
+
+      if(note.readers.indexOf('everyone') == -1){
+        replytoNoteSignatures=[];
+      };
+      console.log('replytoNoteSignatures after filter:',replytoNoteSignatures);
+      var comment_mail = {
+        "groups": replytoNoteSignatures,
+        "subject":"Your post has received a comment",
+        "message": "One of your posts has received a comment.\n\nTitle: "+note.content.title+"\n\nComment: "+note.content.comment+"\n\nTo view the comment, click here: "+baseUrl+"/forum?id=" + note.forum
+      };
+      return or3client.or3request( or3client.mailUrl, comment_mail, 'POST', token );
+    };
+
+    console.log('functions defined')
+    var origNoteP = or3client.or3request(or3client.notesUrl+'?id='+note.forum, {}, 'GET', token);
+    var replytoNoteP = note.replyto ? or3client.or3request(or3client.notesUrl+'?id='+note.replyto,{},'GET',token) : null;
+    console.log('promises created')
+    Promise.all([
+      origNoteP,
+      replytoNoteP
+    ]).then(result => {
+      console.log('promises resolving')
+      var origNote = result[0].notes[0];
+      var origNoteNumber = origNote.number;
+
+      var replytoNote = note.replyto ? result[1].notes[0] : null;
+      var replytoNoteSignatures = replytoNote ? replytoNote.signatures : null;
+
+      console.log('replytoNote',replytoNote);
+      console.log('replytoNoteSignatures',replytoNoteSignatures);
+
+      var promises = [];
+      console.log('note',note)
+      console.log('note.readers',note.readers)
+      var visibleToEveryone = note.readers.indexOf('everyone')>-1 ? true : false;
```

Comment:
Could you use includes? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

```
var visibleToEveryone = note.readers.includes('everyone');
```